pallets/did/src/errors.rs

```diff
@@ -22,54 +22,54 @@ use crate::did_details::DidVerificationKeyRelationship;
 
 /// All the errors that can be generated when validating a DID operation.
 #[derive(Debug, Eq, PartialEq, TypeInfo)]
-pub enum DidError {
+pub enum Error {
 	/// See [StorageError].
-	StorageError(StorageError),
+	Storage(Storage),
 	/// See [SignatureError].
-	SignatureError(SignatureError),
+	Signature(Signature),
 	/// See [InputError].
-	InputError(InputError),
+	Input(Input),
 	/// An error that is not supposed to take place, yet it happened.
-	InternalError,
+	Internal,
 }
 
-impl From<StorageError> for DidError {
-	fn from(err: StorageError) -> Self {
-		DidError::StorageError(err)
+impl From<Storage> for Error {
+	fn from(err: Storage) -> Self {
+		Error::Storage(err)
 	}
 }
 
-impl From<InputError> for DidError {
-	fn from(err: InputError) -> Self {
-		DidError::InputError(err)
+impl From<Input> for Error {
+	fn from(err: Input) -> Self {
+		Error::Input(err)
 	}
 }
 
 /// Error involving the pallet's storage.
 #[derive(Debug, Eq, PartialEq, TypeInfo)]
-pub enum StorageError {
+pub enum Storage {
 	/// The DID being created is already present on chain.
-	DidAlreadyPresent,
+	AlreadyExists,
 	/// The expected DID cannot be found on chain.
-	DidNotPresent,
+	NotFound,
 	/// The given DID does not contain the right key to verify the signature
 	/// of a DID operation.
-	DidKeyNotPresent(DidVerificationKeyRelationship),
+	DidKeyNotFound(DidVerificationKeyRelationship),
```

Comment:
We have `NotFound`, `KeyNotFound` and `DidKeyNotFound`. They all sound very similar.

In case of `DidKeyNotFound`, we can't execute the call since the DID doesn't have the required `VerificationKeyRelationship`? I would interpret that as disallowing the call and rename it to `CallNotAllowed` or similar.
In case of `NotFound` and `KeyNotFound` I would keep `DidNotFound` as it reflects that this is specific for the Did. Best would probably be to have a `NotFound(Resource)` error, where the Resource is an enum that contains all the resources that could be missing.

Reply:
I think now its completely refactored to be in that NotFound(Resource) shape